services/src/handlers/projects.rs

```diff
@@ -13,21 +14,23 @@ pub(crate) fn create_project_handler<C: Context>(
 ) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
     warp::post()
         .and(warp::path("project"))
-        .and(authenticate(ctx))
+        .and(authenticate(ctx.clone()))
+        .and(warp::any().map(move || ctx.clone()))
```

Comment:
Is this okay, because you rely on that C is an `Arc` (cheap to clone) or would it be advantageous to pass references to the actual thing you need, for instance, `&mut ProjectDB`?

Reply:
As far as I seet it I can't pass a reference here because the filter has to capture the variable in order to execute the functions multiple times for multiple different (web) calls.